File: src/patterns/creational/factorymethod/company1.nim
type
    Company1* = ref object
        name: string
        iban: string


proc newCompany1*(name: string, iban: string): Company1 =
    return Company1(
        name: name,
        iban: iban
    )

proc sendMoney*(self: Company1, ibanToSend: string): bool =
    return true

proc getCompanyInfo*(self: Company1): string =
    return "Company Name: " & self.name & "\nIBAN: " & self.iban

proc getName*(self: Company1): string =
    return self.name

proc setName*(self: Company1, name: string) =
    self.name = name

proc getIban*(self: Company1): string =
    return self.iban

proc setIban*(self: Company1, iban: string) =
    self.iban = iban

File: src/patterns/creational/factorymethod/company1.go
package factorymethod


type Company1 struct {
	name string
	iban string
}


func NewCompany1(name string, iban string) *Company1 {
	return &Company1{
		name: name,
		iban:iban,
	}
}

func (self *Company1) SendMoney(ibanToSend string) bool {
	return true
}

func (self *Company1) GetCompanyInfo() string {
	return "Company Name: " + self.name + "\nIBAN: " + self.iban
}

func (self *Company1) GetName() string {
	return self.name
}

func (self *Company1) SetName(name string) {
	self.name = name
}

func (self *Company1) GetIban() string {
	return self.iban
}

func (self *Company1) SetIban(iban string) {
	self.iban = iban
}